appintro/src/main/java/com/github/appintro/AppIntroBase.kt

```diff
@@ -452,18 +452,10 @@ abstract class AppIntroBase : AppCompatActivity(), AppIntroViewPagerListener {
         }
 
         pager.post {
-            val fragment = pagerAdapter.getItem(pager.currentItem)
-            // Fragment is null when no slides are passed to AppIntro
-            if (fragment != null) {
-                dispatchSlideChangedCallbacks(
-                    null,
-                    pagerAdapter
-                        .getItem(pager.currentItem)
-                )
-            } else {
-                // Close the intro if there are no slides to show
-                finish()
-            }
+            dispatchSlideChangedCallbacks(
+                null,
+                pagerAdapter.getItem(pager.currentItem)
+            )
```

Comment:
It seems we don't gracefully handle intros created with 0 slides: in this case, `getItem` will access an out-of-bound element. I thought there was a regression here but the old code was crashing too (there was a useless nullability check since `getItem` was throwing an exception and not a nullable fragment).

I think we can solve this by checking if the current item is in PagerAdapter's bound and, in this case, keep the original behavior (close the intro).

```suggestion
            if (pager.currentItem < pagerAdapter.count) {
                dispatchSlideChangedCallbacks(
                    null,
                    pagerAdapter.getItem(pager.currentItem)
                )
            } else {
                // Close the intro if there are no slides to show
                finish()
            }
```

Reply:
Yup good call 👍